src/main/scala/com/advancedtelematic/campaigner/http/CampaignResource.scala

```diff
@@ -62,11 +62,13 @@ class CampaignResource(extractAuth: Directive1[AuthedNamespaceScope],
     * return them as a CSV file.
     */
   def fetchFailureCodes(ns: Namespace, campaignId: CampaignId, failureCode: String): Route = {
-    val f = campaigns.fetchFailureCodes(campaignId, failureCode).flatMap {
-      Future.traverse(_) { case (did, fc, fd) =>
-        deviceRegistry.fetchOemId(ns, did).map((_, fc, fd))
-      }
-      .map(_.toSeq)
+    val f = campaigns
+      .findLatestFailedUpdates(campaignId, failureCode)
+      .map(_.map(du => (du.device, du.resultCode.getOrElse(""), du.resultDescription.getOrElse(""))))
+      .flatMap {
+        Future.traverse(_) { case (did, fc, fd) =>
+          deviceRegistry.fetchOemId(ns, did).map((_, fc, fd))
```

Comment:
I can imagine following problems with fetching ids individually:

- a lot of requests to device registry
- request times out because it takes too long to complete the future

Reply:
Indeed, we might run into that kind of issues. I proposed a new endpoint in device-registry to fetch all this IDs at once but it wasn't well received. I can create a new ticket to discuss a different approach to improve this. However, this wasn't the bug I was trying to fix in this PR.

Reply:
@koshelev I don't think this ought to be a blocker for this bugfix. Do you think I can merge this?